### PR TITLE
Refactored: Agent now using velocity of kinematic class

### PR DESCRIPTION
**Pull request by Taqi**
Created Kinematic class, and made agent use the velocity of the kinematic class

### CODE REVIEW

File: Assets/Scripts/Flock.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            agent.GetNearbyObjects(neighbourRadius);
+
+            Vector3 velocity = this.behaviour.CalculateMove(agent, agent.Context, this);
+            velocity *= this.driveFactor;
+            if (velocity.sqrMagnitude > this.squareMaxSpeed)
+            {
+                velocity = velocity.normalized * this.maxSpeed;
+            }
+
```
Comment (Taqi) [inline Assets/Scripts/Flock.cs:62]: Setting value of the kinematic velocity here

File: Assets/Scripts/FlockAgent.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public List<Transform> Context { get; set; }
+
+    public Kinematic kinematic;
+
+    void Start()
+    {
+        AgentCollider = GetComponent<Collider>();
+        Context = new List<Transform>();
+
+        kinematic = new Kinematic();
```
Comment (Taqi) [inline Assets/Scripts/FlockAgent.cs:20]: Creating Kinematic object

File: Assets/Scripts/Behaviours/CompositeBehaviour.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                {
+                    partialMove.Normalize();
+                    partialMove *= Weights[i];
+                }
+
+                velocity += partialMove;
+            }
+        }
+
+        return velocity;
```
Comment (Taqi) [inline Assets/Scripts/Behaviours/CompositeBehaviour.cs:37]: Should these behavior method return a Vector3, or should it be refactored and return a Kinematic object instead (like how they did it in the book) ?

File: Assets/Scripts/FlockAgent.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public void Move()
+    {
+        transform.forward = kinematic.linearVel;
+        transform.position += kinematic.linearVel * Time.deltaTime;
```
Comment (Taqi) [inline Assets/Scripts/FlockAgent.cs:26]: Updating position with kinematic velocity

File: Assets/Scripts/Behaviours/SteeredCohesionBehaviour.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+        Vector3 cohesionMove = Vector3.zero;
+        foreach (Transform item in context)
+        {
+            cohesionMove += item.position;
+        }
+
+        cohesionMove /= context.Count;
+        cohesionMove -= agent.transform.position;
+        cohesionMove = Vector3.SmoothDamp(agent.transform.forward, cohesionMove, ref agent.kinematic.linearVel, this.agentSmoothTime);
```
Comment (Taqi) [inline Assets/Scripts/Behaviours/SteeredCohesionBehaviour.cs:23]: referencing the kinematic velocity

File: Assets/Scripts/Flock.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            agent.kinematic.linearVel = velocity;
+            agent.Move();
```
Comment (Taqi) [inline Assets/Scripts/Flock.cs:64]: Setting value of the kinematic velocity here

**Comment by jmcduff** [comment]
I could be wrong, but I'm seeing a couple of files that seem particular to your coding environment (the stuff in the `idea` folders?); could you get those to be ignored by git? Also, could you change the branch you're merging into to Elizabeth's flocking branch? It should make the files changed cleaner and easier to review (we'll change back to master once Elizabeth has merged her branch); this is a good rule to follow whenever you're working on a sub-branch btw.

**Pull request closed without merging**